host-components/README.md

```diff
@@ -0,0 +1,118 @@
+# How to run the host-components
+
+MikeCamel@github.com
+
+2020-08-11
+
+This file explains the steps you need to run the Enarx host-components
+as a Proof of Concept demo.  The exact capabilities of this demo will
+change over time, and so this document should be updated to reflect that.
+
+## What the demo shows
+
+The demo is made up of several components:
+- a keep-manager
+- multiple keep-loaders, created by the by the keep-manager using
+systemctl
+- an app-loader, which waits for, and then runs, a WebAssembly file
+- a keep-manager-tester, which talks to the keep-manager, requesting
+information, setting up the keep-loaders and preparing them to accept
+the WebAssembly file
+- an app-loader-tester, which sends the app-loader a WebAssembly file
+
+## Preparing the demo
+
+ - The demo currently only runs on a single machine, as it listens on
+localhost.  Other hosts are supported, but HTTPS certificates do not work.
+To change the host, edit `LOCAL_LISTEN_ADDRESS` in
+`enarx/host-components/src/lib.rs`.  To test, changes are required in the files
+`enarx/host-components/src/keep-manager-tester.rs` and
+`enarx/host-components/src/app-loader-tester.rs`
+ - The demo currently has no support for Keeps running in TEEs: it supports
+wasmtime.
+ - The demo currently has little support for outputs from a workload.
+ - The demo currently doesn't support filesystem access from a workload.
+ - The demo currently doesn't support socket access from a workload.
+ - The demo currently has a hard-coded .wasm (WebAssembly) file to run.
+ Others can be used, but do not expect much output!
+
+JSON is used for communications.
+Communications to the keep-manager from the keep-manager-tester are over HTTPS.
+Communications to the app-loader from the app-loader-tester are over HTTPS
+with both client and server certificates (the latter being created and self-
+signed by the app-loader - see note above about non-localhost hosts).
+
+In order to run the demo, you shouldn't need any tools beyond those normally
+required to compile Enarx.
+
+Steps:
+1. download the latest source code from github (the easiest way is to do a
+`git clone` of the repository.
```

Comment:
Until this is merged, the easiest way to get this branch locally is to clone locally with:
`git clone --single-branch --branch keep-manager https://github.com/MikeCamel/enarx/`